clang/lib/Driver/Driver.cpp

```diff
@@ -3466,21 +3489,27 @@ class OffloadingActionBuilder final {
 
         // After the Link, wrap the files before the final host link
         if (SYCLAOTCompile) {
-          types::ID OutType = types::TY_Image;
-          if (TT.getSubArch() == llvm::Triple::SPIRSubArch_fpga)
-            OutType = FPGAOutType;
+          types::ID OutType;
+          if (NoDeviceCodeSplit) {
+            OutType = types::TY_Image;
+            if (TT.getSubArch() == llvm::Triple::SPIRSubArch_fpga)
+              OutType = FPGAOutType;
```

Comment:
Would a ternary suit this particular condition better?

Reply:
Maybe. Check if it feels better.

Reply:
A tiny bit :) Thanks.